roles/openshift_certificate_expiry/test/test_fakeopensslclasses.py

```diff
@@ -1,82 +1,78 @@
+    ''' Validate the certificate subject '''
 
-        self.assertEqual('DNS:kubernetes, DNS:kubernetes.default, DNS:kubernetes.default.svc, DNS:kubernetes.default.svc.cluster.local, DNS:m01.example.com, DNS:openshift, DNS:openshift.default, DNS:openshift.default.svc, DNS:openshift.default.svc.cluster.local, DNS:172.30.0.1, DNS:192.168.122.241, IP Address:172.30.0.1, IP Address:192.168.122.241', sans)
+    # Gather the subject components and create a list of colon separated strings.
+    # Since the internal representation of pyOpenSSL uses bytes, we need to decode
+    # the results before comparing.
+    c_subjects = valid_cert['cert'].get_subject().get_components()
+    c_subj = ', '.join(['{}:{}'.format(x.decode('utf8'), y.decode('utf8')) for x, y in c_subjects])
+    f_subjects = fake_valid_cert.get_subject().get_components()
+    f_subj = ', '.join(['{}:{}'.format(x, y) for x, y in f_subjects])
+    assert c_subj == f_subj
 
-    def test_FakeOpenSSLCertificate_get_sans_no_sans(self):
-        """We can tell when there are no Subject Alt Names in a cert"""
-        with self.assertRaises(IndexError):
-            self.fake_cert.get_extension(0)
 
-    def test_FakeOpenSSLCertificate_get_subject(self):
-        """We can read the Subject from a cert"""
-        # Subject: O=system:nodes, CN=system:node:m01.example.com
-        subject = self.fake_cert.get_subject()
-        subjects = []
-        for name, value in subject.get_components():
-            subjects.append('{}={}'.format(name, value))
+def test_subject_alt_names(valid_cert, fake_valid_cert):
+    real_cert = valid_cert['cert']
 
-        self.assertEqual('O=system:nodes, CN=system:node:m01.example.com', ', '.join(subjects))
-
-    def test_FakeOpenSSLCertificate_get_subject_san_cert(self):
-        """We can read the Subject from a cert with sans"""
-        # Subject: O=system:nodes, CN=system:node:m01.example.com
-        subject = self.fake_san_cert.get_subject()
-        subjects = []
-        for name, value in subject.get_components():
-            subjects.append('{}={}'.format(name, value))
+    # Gather the string representation of the subjectAltName
+    # extensions, and verify they match
+    san = None
+    for i in range(real_cert.get_extension_count()):
+        ext = real_cert.get_extension(i)
+        # pyOpenSSL uses bytes for the internal representation
+        if ext.get_short_name() == b'subjectAltName':
+            san = str(ext)
+    f_san = None
+    for i in range(fake_valid_cert.get_extension_count()):
+        ext = fake_valid_cert.get_extension(i)
+        if ext.get_short_name() == 'subjectAltName':
+            f_san = str(ext)
+    assert san == f_san
 
-        self.assertEqual('CN=172.30.0.1', ', '.join(subjects))
+    # If there are either dns or ip sans defined, verify common_name present
+    if valid_cert['ip'] or valid_cert['dns']:
+        assert 'DNS:{}'.format(valid_cert['common_name']) in f_san
```

Comment:
In this particular case where the format string is just a prefix, wouldn't it be better to write it like below?

```diff
-assert 'DNS:{}'.format(valid_cert['common_name']) in f_san
+assert 'DNS:' + valid_cert['common_name'] in f_san
```